content/posts/0008-introducing-liberty.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Introducing the Liberty buildpack
+date: "2022-04-04T09:38:38-04:00"
+slug: introducing-liberty-buildpack
+author: kevin-ortega
+---
+
+![Open Liberty Logo](/images/posts/0008/open-liberty-logo.png)
+
+[Liberty](https://github.com/paketo-buildpacks/liberty) is finally available in Paketo and included in the [java](https://github.com/paketo-buildpacks/java) composite buildpack! Liberty is an open application framework designed for the cloud. It’s small, lightweight, and designed with modern cloud-native application development in mind.
+
+With the Liberty buildpack, you can:
+*   Build java apps from source or a pre-configured on-prem Liberty
+*   Install Liberty and user custom features
+*   Install Liberty interim fixes
+*   Create UBI-based OCI images
+
+## What do you need?
+*   Your application source or a pre-configured on-prem Liberty
+*   [Docker](https://hub.docker.com/search?type=edition&offering=community)
+*   [pack](https://buildpacks.io/docs/tools/pack/)
+
+## Let's get started!
+
+The following examples use the [Open Liberty starter](https://openliberty.io/guides/getting-started.html) application as the application source.
+Download the Open Liberty starter application:
+```
+git clone https://github.com/openliberty/guide-getting-started.git
+cd guide-getting-started/start
+```
+
+Set a default builder:
+```
+pack config default-builder gcr.io/paketo-buildpacks/builder:base
+```
+
+Build the application with a minimal footprint with only the Liberty features required to run the application and IBM Semeru OpenJ9:
+```
+ pack build --env BP_JAVA_APP_SERVER=liberty --env BP_LIBERTY_PROFILE=kernel \  
+  --env BP_LIBERTY_FEATURES='jaxrs-2.1 jsonp-1.1 cdi-2.0 mpMetrics-3.0 mpConfig-2.0' \
+  --buildpack paketo-buildpacks/eclipse-openj9 --buildpack paketo-buildpacks/java myapp
+```
+
+Your application is now transformed into an OCI image!
+
+## Now what?
+With your OCI image, you can run your application locally with the `docker run` command.
+
+We need to provide a server.xml to the Liberty buildpack using a binding.  
+1. Create a directory named `bindings`, for example
+2. Create a file named server.xml in the `bindings` directory with the following content:
+```
+<server description="Intro to the Paketo Liberty buildpack">
+  <!-- tag::featureManager[] -->
+  <featureManager>
+      <!--feature>webProfile-8.0</feature -->
+      <feature>mpconfig-2.0</feature>
+      <feature>mpmetrics-3.0</feature>
+      <feature>cdi-2.0</feature>
+      <feature>jsonp-1.1</feature>
+      <feature>jaxrs-2.1</feature>
+  </featureManager>
+  <!-- end::featureManager[] -->
+
+  <mpMetrics authentication="false"/>
+
+  <!-- tag::httpEndpoint[] -->
+  <httpEndpoint httpPort="9080" httpsPort="9443" id="defaultHttpEndpoint" host="*" />
+  <!-- end::httpEndpoint[] -->
+</server>
+```
```

Comment:
Is this strictly necessary? I'm pretty sure that we're writing a server.xml that has the features enabled, when you set `BP_LIBERTY_FEATURES` and the `httpEndpoint` should be set by default as well.

Reply:
Yeah, the httpEndpoint is probably not necessary but the list of features is because we are not currently writing the server.xml with the features when you set the variable.  That is something we intend to address.

Reply:
👍  OK, no worries. We can get that fixed up later & we can update the blog post when it's ready.